types/cache.go

```diff
@@ -25,3 +25,16 @@ type SavingsRequest struct {
 	GradleMetrics gradle.Metrics `json:"gradle_metrics"`
 	DlcMetrics    dlc.Metrics    `json:"dlc_metrics"`
 }
+
+type SavingsOverview struct {
+	FeatureName  SavingsFeature             `json:"feature_name"`
+	TimeTakenMs  int64                      `json:"time_taken_ms"`
+	TimeSavedMs  int64                      `json:"time_saved_ms"`
+	BaselineMs   int64                      `json:"baseline_ms"`
+	FeatureState IntelligenceExecutionState `json:"feature_state"`
+}
+
+type SavingsResponse struct {
+	Overview   []SavingsOverview `json:"overview"`
```

Comment:
This is an array to accommodate for stage-level requests?

Reply:
Nope. One step can have multiple Intelligence feature. Each array element is for the details of a particular feature